datafusion/functions-window/src/ntile.rs

```diff
@@ -0,0 +1,192 @@
+use datafusion_physical_expr_common::physical_expr::PhysicalExpr;
+use field::WindowUDFFieldArgs;
+
+get_or_init_udwf!(
+    Ntile,
+    ntile,
+    "integer ranging from 1 to the argument value, dividing the partition as equally as possible"
+);
+
+pub fn ntile(arg: i64) -> datafusion_expr::Expr {
+    ntile_udwf().call(vec![arg.lit()])
+}
+
+#[derive(Debug)]
+pub struct Ntile {
+    signature: Signature,
+}
+
+impl Ntile {
+    /// Create a new `ntile` function
+    pub fn new() -> Self {
+        Self {
+            signature: Signature::uniform(
+                1,
+                vec![
+                    DataType::UInt64,
+                    DataType::UInt32,
+                    DataType::UInt16,
+                    DataType::UInt8,
+                    DataType::Int64,
+                    DataType::Int32,
+                    DataType::Int16,
+                    DataType::Int8,
+                ],
+                Volatility::Immutable,
+            ),
+        }
+    }
+}
+
+impl Default for Ntile {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+static DOCUMENTATION: OnceLock<Documentation> = OnceLock::new();
+
+fn get_ntile_doc() -> &'static Documentation {
+    DOCUMENTATION.get_or_init(|| {
+        Documentation::builder()
+            .with_doc_section(DOC_SECTION_RANKING)
+            .with_description(
+                "Integer ranging from 1 to the argument value, dividing the partition as equally as possible",
+            )
+            .with_syntax_example("ntile(expression)")
+            .with_argument("expression","An integer describing the number groups the partition should be split into")
+            .build()
+            .unwrap()
+    })
+}
+
+impl WindowUDFImpl for Ntile {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn name(&self) -> &str {
+        "ntile"
+    }
+
+    fn signature(&self) -> &Signature {
+        &self.signature
+    }
+
+    fn expressions(&self, expr_args: ExpressionArgs) -> Vec<Arc<dyn PhysicalExpr>> {
+        parse_expr(expr_args.input_exprs(), expr_args.input_types())
+            .into_iter()
+            .collect::<Vec<_>>()
+    }
+
+    fn partition_evaluator(
+        &self,
+        partition_evaluator_args: PartitionEvaluatorArgs,
+    ) -> Result<Box<dyn PartitionEvaluator>> {
+        let scalar_n =
+            get_scalar_value_from_args(partition_evaluator_args.input_exprs(), 0)?
+                .ok_or_else(|| {
+                    DataFusionError::Execution(
+                        "NTILE requires a positive integer".to_string(),
+                    )
+                })?;
+
+        if scalar_n.is_null() {
+            return exec_err!("NTILE requires a positive integer, but finds NULL");
+        }
+
+        let n = get_unsigned_integer(scalar_n)?;
+        if n == 0 {
+            return exec_err!("NTILE requires a positive integer");
+        }
+
+        Ok(Box::new(NtileEvaluator { n }))
```

Comment:
Let's keep the original code from `create_built_in_window_expr` for parsing `n` unmodified?

```suggestion
            if scalar_n.is_unsigned() {
                let n = get_unsigned_integer(scalar_n)?;
                Ok(Box::new(NtileEvaluator { n }))
            } else {
                let n: i64 = get_signed_integer(scalar_n)?;
                if n <= 0 {
                    return exec_err!("NTILE requires a positive integer");
                }
                Ok(Box::new(NtileEvaluator { n: n as u64 }))
            }
```